### PR TITLE
build(client): Update eslint config to latest version (5.6.0)

### PR DESCRIPTION
**Pull request by tylerbutler**
Updates the client release group to eslint config 5.6.0.

### CODE REVIEW

File: packages/service-clients/end-to-end-tests/azure-client/.eslintrc.cjs [hunk 1]
```diff
@@ -8,6 +8,7 @@ module.exports = {
 	rules: {
 		"prefer-arrow-callback": "off",
 		"@typescript-eslint/strict-boolean-expressions": "off", // requires strictNullChecks=true in tsconfig
+		"@fluid-internal/fluid/no-unchecked-record-access": "warn",
```
Comment (alexvy86) [inline packages/service-clients/end-to-end-tests/azure-client/.eslintrc.cjs:11]: TODO or comment as to why we disable it?

File: packages/test/stochastic-test-utils/.eslintrc.cjs [hunk 2]
```diff
@@ -12,6 +12,7 @@ module.exports = {
 		project: ["./tsconfig.json", "./src/test/tsconfig.json"],
 	},
 	rules: {
+		"@fluid-internal/fluid/no-unchecked-record-access": "warn",
```
Comment (alexvy86) [inline packages/test/stochastic-test-utils/.eslintrc.cjs:15]: Idem

File: packages/test/test-pairwise-generator/.eslintrc.cjs [hunk 3]
```diff
@@ -5,4 +5,7 @@
 
 module.exports = {
 	extends: ["@fluidframework/eslint-config-fluid/minimal-deprecated", "prettier"],
+	rules: {
+		"@fluid-internal/fluid/no-unchecked-record-access": "warn",
```
Comment (alexvy86) [inline packages/test/test-pairwise-generator/.eslintrc.cjs:9]: One more

File: packages/tools/devtools/devtools-browser-extension/.eslintrc.cjs [hunk 4]
```diff
@@ -26,6 +26,8 @@ module.exports = {
 				devDependencies: ["src/**/test/**"],
 			},
 		],
+
+		"@fluid-internal/fluid/no-unchecked-record-access": "warn",
```
Comment (alexvy86) [inline packages/tools/devtools/devtools-browser-extension/.eslintrc.cjs:30]: And one more

**Review by alexvy86 — APPROVED** [review]
A couple of comments. The overrides in PropertyDDS I can understand without additional comments, but the other ones feel like they should have something.

**Comment by tylerbutler** [comment]
@RishhiB is actively working on turning these back on.

File: experimental/PropertyDDS/packages/property-changeset/.eslintrc.cjs
Comment (jason-ha) [inline experimental/PropertyDDS/packages/property-changeset/.eslintrc.cjs:1]: Is this change needed? @RishhiB, did some needed changes get missed in your update? Or is this PR hitting more than required?

File: experimental/PropertyDDS/packages/property-properties/.eslintrc.cjs
Comment (jason-ha) [inline experimental/PropertyDDS/packages/property-properties/.eslintrc.cjs:1]: I am assuming at least this config change remains if the no-unchecked-record-access additions are reverted.
Please add notes to the change description.

File: experimental/PropertyDDS/packages/property-common/.eslintrc.cjs [hunk 5]
```diff
@@ -12,6 +12,7 @@ module.exports = {
 		project: ["./tsconfig.json", "./src/test/tsconfig.json"],
 	},
 	rules: {
+		"@fluid-internal/fluid/no-unchecked-record-access": "warn",
 		"prefer-arrow-callback": "off",
```
Comment (jason-ha) [inline experimental/PropertyDDS/packages/property-common/.eslintrc.cjs:16]: Re-issue post conflict resolution:
Is this and similar changes needed? @RishhiB, did some needed changes get missed in your update? Or is this PR hitting more than required?

Comment (tylerbutler) [reply experimental/PropertyDDS/packages/property-common/.eslintrc.cjs:16]: In my testing, all of them have been necessary to pass CI. There might have been new code introduced or they somehow got missed. @RishhiB is pulling the config changes out into a separate PR. I'll mark this as a draft in the meantime.
